chanzuckerberg/idseq-web
# [Data Discovery] Fix samples and project counters out of sync issues

# Description

* Fixes bugs when counter were not updating after some filters were set
* Enforce a behavior where a location in the map is deselected if the user selects a filter with no results while the location is selected
	- We might re-evaluate this behavior later
* Added extra comments to Discovery Data Layer.

# Tests

1. Navigate to the My Data (or other domain) and projects tab
2. Set a few filters
	* Verify that counters update accordingly for both projects and samples on top nav
	* Verify that counters update accordingly for both projects and samples on sidebar
3. Navigate to samples tabs
4. Change/Add/Remove filters
	* Verify that counters update accordingly for both projects and samples on top nav
	* Verify that counters update accordingly for both projects and samples on sidebar
5. Change to map view
6. Click on a particular location
	* Verify that counters remain the same in the top nav 
	* Verify that counters update accordingly for both projects and samples on sidebar
7. Deselect the location by clicking anywhere on the map
	* Verify that counters remain the same in the top nav 
	* Verify that counters go back to the previous values
